src/Module/Settings/Uexport.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * @file src/Modules/Settings/Uexport.php
+ */
+
+namespace Friendica\Module\Settings;
+
+use Friendica\App;
+use Friendica\App\Arguments;
+use Friendica\BaseModule;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\System;
+use Friendica\Database\DBA;
+use Friendica\Database\DBStructure;
+use Friendica\Module\BaseSettingsModule;
+
+/**
+ * Module to export user data
+ **/
+class Uexport extends BaseSettingsModule
+{
+	/**
+	 * Handle the request to export data.
+	 * At the moment one can export two different data set
+	 * 1. The profile data that can be used by uimport to resettle
+	 *    to a different Friendica instance
+	 * 2. The entire data-set, profile plus postings
+	 *
+	 * If there is an action required through the URL / path, react
+	 * accordingly and export the requested data.
+	 **/
+	public static function content()
+	{
+		parent::content();
+		$args = self::getClass(Arguments::class);
+		if ($args->getArgc() == 3) {
+			// @TODO Replace with router-provided arguments
+			$action = $args->get(2);
+			header("Content-type: application/json");
+			header('Content-Disposition: attachment; filename="' . $a->user['nickname'] . '.' . $action . '"');
+			switch ($action) {
+				case "backup":
+					SELF::uexport_all($a);
+					exit();
+					break;
+				case "account":
+					SELF::uexport_account($a);
+					exit();
+					break;
+				default:
+					exit();
+			}
+		}
+
+		/**
+		 * options shown on "Export personal data" page
+		 * list of array( 'link url', 'link text', 'help text' )
+		 */
+		$options = [
+			['settings/uexport/account', L10n::t('Export account'), L10n::t('Export your account info and contacts. Use this to make a backup of your account and/or to move it to another server.')],
+			['settings/uexport/backup', L10n::t('Export all'), L10n::t("Export your accout info, contacts and all your items as json. Could be a very big file, and could take a lot of time. Use this to make a full backup of your account \x28photos are not exported\x29")],
+		];
+		Hook::callAll('uexport_options', $options);
+
+		$tpl = Renderer::getMarkupTemplate("uexport.tpl");
+		return Renderer::replaceMacros($tpl, [
+			'$title' => L10n::t('Export personal data'),
+			'$options' => $options
+		]);
+	}
+	private function uexport_multirow($query) {
+		global $dbStructure;
+
+		preg_match("/\s+from\s+`?([a-z\d_]+)`?/i", $query, $match);
+		$table = $match[1];
+
+		$result = [];
+		$r = q($query);
```

Comment:
plz use `DBA::p()` instead. Returns the same result

Reply:
And again the question, why the original module was not adopted to the new query style before. I remember those mass PRs that just contained replacements for those queries.

Reply:
The mass-replaced queries were the most straighforward. Replacing `q()` calls is a case-by-case basis as `%s` argument placeholders have to be replaced by `?` for example. In this case however since it doesn't use arguments in the `q()` call it should be a simple replacement.